### ISSUE TITLE
MDSIP connection waits should handle aborts/SIGINT

### ISSUE THREAD

**Issue by zack-vii** [MDSplus/mdsplus#1362]
currently if you do e.g.
```tcl
show server myserver:port
```
The request times out after several seconds if the  "Transport endpoint is not connected".
I those cases it would be great if one could either set the timeout or abort the attempt via SIGINT (Ctrl+C).

The service should be able to handle such aborted waits/requests merciful and eventually abort any further execution of the dispatched command.

**Comment by tfredian** [comment]
Providing SIGINT handling of the show server command is easy to do and I tested an implementation of it. Aborting a executing command on the server if it is not responding to a show server command is another story which might be extremely difficult to implement since the mdsip process on the server must be in some unresponsive state in the first place. If show server does not get a response from the server the action it is performing must be preventing the main mdsip process thread from receiving connections or executing expressions sent from the client since that is how show server works. This would indicate a problem with the command being executed on the server which should be investigated.
It is unlikely the mdsip process would respond to any message sent to the server over the port the server mdsip process is listening so even abort or stop messages would be blocked as well. One could imagine implementing a tdi function that finds the pid of the server process and sends it a signal so that the client could make an mdsip connection using a different port to kill or interrupt the mdsip action server but that would require the use of an account with sufficient privileges.

So... should I implement the SIGINT interruption of the show server command which would only affect the client mdstcl process and have no affect on the hung server process?

**Comment by joshStillerman** [comment]
Is this now resolved then ?

**Comment by zack-vii** [comment]
does not seem to be the case
```tdi
show server 8.8.8.8:80
Checking server: 8.8.8.8:80
^C^C^C^C^CError in connect to service: Transport endpoint is not connected
Error connecting to server
```

**Comment by zack-vii** [comment]
could eventually be fixed in the process of #1443

**Comment by zack-vii** [comment]
fixed by #1781
